mirascope/core/vertex/call_response.py

```diff
@@ -0,0 +1,177 @@
+"""This module contains the `VertexCallResponse` class.
+
+usage docs: learn/calls.md#handling-responses
+"""
+
+from google.cloud.aiplatform_v1beta1.types import (
+    tool as gapic_tool_types,
+)
+from pydantic import computed_field
+from vertexai.generative_models import (
+    Content,
+    GenerationResponse,
+    Tool,
+)
+
+from ..base import BaseCallResponse
+from ._utils import calculate_cost
+from .call_params import VertexCallParams
+from .dynamic_config import VertexDynamicConfig
+from .tool import VertexTool
+
+
+class VertexCallResponse(
+    BaseCallResponse[
+        GenerationResponse,
+        VertexTool,
+        Tool,
+        VertexDynamicConfig,
+        Content,
+        VertexCallParams,
+        Content,
+    ]
+):
+    """Convenience wrapper around Vertex's `GenerateContentResponse`.
+
+    When using Mirascope's convenience wrappers to interact with Vertex models via
+    `VertexCall`, responses using `VertexCall.call()` will return a
+    `VertexCallResponse`, whereby the implemented properties allow for simpler syntax
+    and a convenient developer experience.
+
+    Example:
+
+    ```python
+    from mirascope.core import prompt_template
+    from mirascope.core.vertex import vertex_call
+
+
+    @vertex_call("gemini-1.5-flash")
+    @prompt_template("Recommend a {genre} book")
+    def recommend_book(genre: str):
+        ...
+
+
+    response = recommend_book("fantasy")  # response is an `VertexCallResponse` instance
+    print(response.content)
+    ```
+    """
+
+    _provider = "vertex"
+
+    @property
+    def content(self) -> str:
+        """Returns the contained string content for the 0th choice."""
+        return self.response.candidates[0].content.parts[0].text
+
+    @property
+    def finish_reasons(self) -> list[str]:
+        """Returns the finish reasons of the response."""
+        finish_reasons = [
+            "FINISH_REASON_UNSPECIFIED",
+            "STOP",
+            "MAX_TOKENS",
+            "SAFETY",
+            "RECITATION",
+            "OTHER",
+        ]
+
+        return [
+            finish_reasons[candidate.finish_reason]
+            for candidate in self.response.candidates
+        ]
+
+    @property
+    def model(self) -> str:
+        """Returns the model name.
+
+        google.generativeai does not return model, so we return the model provided by
+        the user.
+        """
+        return self._model
+
+    @property
+    def id(self) -> str | None:
+        """Returns the id of the response.
+
+        google.generativeai does not return an id
+        """
+        return None
+
+    @property
+    def usage(self) -> None:
+        """Returns the usage of the chat completion.
+
+        google.generativeai does not have Usage, so we return None
```

Comment:
nit: same here, also I think vertex returns usage, so this may actually need to return something other than `None`